src/battle-dex-search.ts

```diff
@@ -1202,6 +1202,10 @@ class BattleMoveSearch extends BattleTypedSearch<'move'> {
 			return !moves.includes('energyball') && !moves.includes('gigadrain');
 		case 'hiddenpowerice':
 			return !moves.includes('icebeam') && !moves.includes('aurorabeam');
+		case 'hiddenpowerflying':
+			return (species.baseSpecies === 'Gyarados' || species.baseSpecies === 'Aerodactyl' || species.baseSpecies === 'Salamence' || species.baseSpecies === 'Charizard' || species.baseSpecies === 'Crobat' || species.baseSpecies === 'Dragonite' || species.baseSpecies === 'Ninjask' || species.baseSpecies === 'Swellow' || species.baseSpecies === 'Gligar') && dex.gen < 4;
+		case 'hiddenpowerbug':
+			return (species.baseSpecies === 'Forretress' || species.baseSpecies === 'Pinsir' || species.baseSpecies === 'Armaldo' || species.baseSpecies === 'Ninjask' || species.baseSpecies === 'Tyranitar' || species.baseSpecies === 'Scizor' || species.baseSpecies === 'Aerodactyl' || species.baseSpecies === 'Dugtrio' || species.baseSpecies === 'Flygon' || species.baseSpecies === 'Marowak' || species.baseSpecies === 'Machamp' || species.baseSpecies === 'Sandslash') && dex.gen < 4;
```

Comment:
```suggestion
    case 'hiddenpowerflying':
        return dex.gen < 4 && [
            'Gyarados', 'Aerodactyl', 'Salamence', 'Charizard', 'Crobat', 'Dragonite', 'Ninjask', 'Swellow', 'Gligar'
        ].includes(species.baseSpecies);
    case 'hiddenpowerbug':
        return dex.gen < 4 && [
            'Forretress', 'Pinsir', 'Armaldo', 'Ninjask', 'Tyranitar', 'Scizor', 
            'Aerodactyl', 'Dugtrio', 'Flygon', 'Marowak', 'Machamp', 'Sandslash'
        ].includes(species.baseSpecies);
```
I'm not very familiar with this code, so there might be some other issue I'm missing here, but you probably should just use arrays here.

Reply:
We tend not to hardcode species like this.

Here's how I'd do it:

```suggestion
		case 'hiddenpowerflying':
			return dex.gen < 4;
		case 'hiddenpowerbug':
			return dex.gen < 4 && !moves.includes('megahorn');
```